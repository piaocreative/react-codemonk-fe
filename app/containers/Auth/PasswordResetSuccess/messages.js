/*
 * PasswordResetSuccess Messages
 *
 * This contains all the text for the PasswordResetSuccess component.
 */
import { defineMessages } from 'react-intl';

export const scope = 'CodeMonk.components.PasswordResetSuccess';

export default defineMessages({
  title: {
    id: `${scope}.title.head`,
    defaultMessage: 'Password Changed!',
  },
  metaTitle: {
    id: `${scope}.metaTitle.head`,
    defaultMessage: 'Password Changed!',
  },
  headingPasswordChanged: {
    id: `${scope}.headingPasswordChanged`,
    defaultMessage: 'Password Changed!',
  },
  textIntro: {
    id: `${scope}.textIntro`,
    defaultMessage: 'Your password has successfully been changed! Login to your account.',
  },
  signInButtonText: {
    id: `${scope}.signInButtonText`,
    defaultMessage: 'Login',
  },
});
