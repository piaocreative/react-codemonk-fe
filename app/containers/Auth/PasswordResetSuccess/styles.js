import styled from 'styled-components';
import SVG from 'react-inlinesvg';

export const IntroText = styled.p`
  max-width: 790px;
  margin: 0 auto;
`;

export const TickIcon = styled(SVG)`
  width: 40px;
  height: 40px;
`;
