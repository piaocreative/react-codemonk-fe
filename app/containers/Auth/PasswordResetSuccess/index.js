/**
 * PasswordResetSuccess
 * This is the PasswordResetSuccess for the App, at the '/password-reset-success' route
 */

import React, { Component } from 'react';
import { Helmet } from 'react-helmet';
import { FormattedMessage } from 'react-intl';
import StorageService from 'utils/StorageService';
import { defaultProps, propTypes } from 'containers/proptypes';
import { OnboardingForm, P, H1, OnBoardingFormBody, ContainerMod, Card } from 'components';
import { loginLink } from 'containers/App/utils';
import { LinkMod } from 'components/A';
import { circleTickIcon } from 'containers/App/constants';
import messages from './messages';
import { TickIcon } from './styles';

export class PasswordResetSuccess extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isAgency: false,
    };
  }

  componentDidMount() {
    this.setSelectedRole();
  }

  setSelectedRole = () => {
    const { location } = this.props;
    if (location.pathname.split('/').includes('agency')) {
      StorageService.set('selectedRoleType', 'agency', { hash: true });
      this.setState({ isAgency: true });
    } else {
      StorageService.set('selectedRoleType', 'talent', { hash: true });
    }
  };

  componentWillReceiveProps(nextProps) {
    if (nextProps.location.pathname.split('/').includes('agency')) {
      StorageService.set('selectedRoleType', 'agency', { hash: true });
      this.setState({ isAgency: true });
    } else {
      StorageService.set('selectedRoleType', 'talent', { hash: true });
    }
  }

  render() {
    const { isAgency } = this.state;

    return (
      <>
        <Helmet>
          <title>{messages.title.defaultMessage}</title>
          <meta name="description" content={messages.metaTitle.defaultMessage} />
        </Helmet>
        <ContainerMod className={!isAgency ? 'w-auto p-0' : ''}>
          <Card className={isAgency ? 'text-center' : 'p-0 bg-transparent'}>
            {isAgency && (
              <>
                <TickIcon className="mb-4 mb-md-5 w-1" src={circleTickIcon} />
                <H1 className="mb-3">
                  <FormattedMessage {...messages.headingPasswordChanged} />
                </H1>
                <P>
                  <FormattedMessage {...messages.textIntro} />
                </P>
              </>
            )}
            <OnboardingForm>
              <OnBoardingFormBody>
                {!isAgency && (
                  <>
                    <TickIcon className="mb-4 mb-md-5 w-1" src={circleTickIcon} />
                    <H1 className="mb-3">
                      <FormattedMessage {...messages.headingPasswordChanged} />
                    </H1>
                    <P className="p16" opacityVal="0.5">
                      <FormattedMessage {...messages.textIntro} />
                    </P>
                  </>
                )}
                {!isAgency && (
                  <LinkMod to={loginLink} className="btn-primary w-100 mt-3">
                    <FormattedMessage {...messages.signInButtonText} />
                  </LinkMod>
                )}
              </OnBoardingFormBody>
              {isAgency && (
                <>
                  <hr />
                  <OnBoardingFormBody className="d-flex justify-content-center">
                    <LinkMod to="/agency/login" className="btn-primary">
                      <FormattedMessage {...messages.signInButtonText} />
                    </LinkMod>
                  </OnBoardingFormBody>
                </>
              )}
            </OnboardingForm>
          </Card>
        </ContainerMod>
      </>
    );
  }
}

PasswordResetSuccess.defaultProps = defaultProps;
PasswordResetSuccess.propTypes = propTypes;

export default PasswordResetSuccess;
