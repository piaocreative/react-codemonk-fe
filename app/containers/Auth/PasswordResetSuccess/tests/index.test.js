import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import PasswordResetSuccess from '../index';

describe('<MainForm />', () => {
  const renderer = new ShallowRenderer();
  it('should render and match the snapshot', () => {
    renderer.render(<PasswordResetSuccess />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
