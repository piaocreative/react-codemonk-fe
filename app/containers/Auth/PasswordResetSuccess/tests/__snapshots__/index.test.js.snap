// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainForm /> should render and match the snapshot 1`] = `
<React.Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Password Changed!
    </title>
    <meta
      content="Password Changed!"
      name="description"
    />
  </HelmetWrapper>
  <ForwardRef(ContainerMod)
    className="w-auto p-0"
  >
    <ForwardRef(Card)
      className="p-0 bg-transparent"
    >
      <ForwardRef(OnboardingForm)>
        <ForwardRef(OnboardingForm__OnBoardingFormBody)>
          <React.Fragment>
            <ForwardRef(styles__TickIcon)
              className="mb-4 mb-md-5 w-1"
              src="IMAGE_MOCK"
            />
            <ForwardRef(H1)
              className="mb-3"
            >
              <FormattedMessage
                defaultMessage="Password Changed!"
                id="CodeMonk.components.PasswordResetSuccess.headingPasswordChanged"
                values={Object {}}
              />
            </ForwardRef(H1)>
            <ForwardRef(P)
              className="p16"
              opacityVal="0.5"
            >
              <FormattedMessage
                defaultMessage="Your password has successfully been changed! Login to your account."
                id="CodeMonk.components.PasswordResetSuccess.textIntro"
                values={Object {}}
              />
            </ForwardRef(P)>
          </React.Fragment>
          <ForwardRef(A__LinkMod)
            className="btn-primary w-100 mt-3"
            to="/login"
          >
            <FormattedMessage
              defaultMessage="Login"
              id="CodeMonk.components.PasswordResetSuccess.signInButtonText"
              values={Object {}}
            />
          </ForwardRef(A__LinkMod)>
        </ForwardRef(OnboardingForm__OnBoardingFormBody)>
      </ForwardRef(OnboardingForm)>
    </ForwardRef(Card)>
  </ForwardRef(ContainerMod)>
</React.Fragment>
`;
